include/fiction/algorithms/simulation/sidb/is_operational.hpp

```diff
@@ -387,72 +474,179 @@ class is_operational_impl
                 // number of different input combinations
                 for (auto i = 0u; i < truth_table.front().num_bits(); ++i, ++bii)
                 {
-                    if (is_layout_invalid(bii.get_current_input_index(), cds_canvas))
+                    if (is_layout_invalid(bii.get_current_input_index()))
                     {
-                        return {operational_status::NON_OPERATIONAL, non_operationality_reason::LOGIC_MISMATCH};
+                        return {operational_assessment<Lyt>{operational_status::NON_OPERATIONAL},
+                                non_operationality_reason::LOGIC_MISMATCH};
                     }
                 }
             }
+
+            // if the layout is not discarded during the three filtering steps, it is considered operational.
+            // This is only an approximation.
+            if (parameters.strategy_to_analyze_operational_status ==
+                is_operational_params::operational_analysis_strategy::FILTER_ONLY)
+            {
+                return {operational_assessment<Lyt>{operational_status::OPERATIONAL}, non_operationality_reason::NONE};
+            }
+
+            if (parameters.strategy_to_analyze_operational_status !=
+                    is_operational_params::operational_analysis_strategy::SIMULATION_ONLY &&
+                parameters.strategy_to_analyze_operational_status !=
+                    is_operational_params::operational_analysis_strategy::FILTER_THEN_SIMULATION)
+            {
+                return {operational_assessment<Lyt>{operational_status::OPERATIONAL}, non_operationality_reason::NONE};
+            }
         }
 
-        // if the layout is not discarded during the three filtering steps, it is considered operational.
-        // This is only an approximation.
-        if (parameters.strategy_to_analyze_operational_status ==
-                is_operational_params::operational_analysis_strategy::FILTER_ONLY &&
-            !canvas_lyt.is_empty())
+        operational_assessment<Lyt> assessment_results{operational_status::OPERATIONAL};
```

Comment:
```suggestion
        operational_assessment<Lyt> operational_assessment_results{operational_status::OPERATIONAL};
```
I think this would improve the readability.